spec/tapioca/compilers/symbol_table_compiler_spec.rb

```diff
@@ -2719,6 +2719,98 @@ def inherited(s); end
       assert_equal(output, compile)
     end
 
+    it("compiles nested generic interfaces") do
+      add_ruby_file("generic.rb", <<~RUBY)
+        module Root
+          extend T::Sig
+          extend T::Generic
+          Elem = type_member
+
+          interface!
+
+          sig { abstract.returns(T::Array[Node[Elem]]) }
+          def children; end
```

Comment:
When something `include Root` and redefines this method, tapioca starts producing the following sig:
```
sig { abstract.returns(T::Array[Node[Elem]]) }
```